### PR TITLE
Track card metrics for SwuStats integration

### PR DESCRIPTION
**Pull request by Frugghi**

### CODE REVIEW

File: test/scenarios/undo/Undo.spec.ts [hunk 1]
```diff
@@ -1765,12 +1765,12 @@ describe('Undo', function() {
...
+                        resources: 6,
                         deck: [],
                     },
                     player2: {
                         hand: ['crafty-smuggler'],
-                        resources: 2,
+                        resources: 6,
                         deck: []
                     }
                 });
```
Comment (AMMayberry1) [inline test/scenarios/undo/Undo.spec.ts:1776]: Can we add one undo test just to validate the behavior and establish a pattern for how to test the stats stuff. I think we'll need a few tests of edge cases involving stats + undo in the near future

File: server/game/core/Player.ts [hunk 2]
```diff
@@ -130,11 +131,11 @@ export class Player extends GameObject<IPlayerState> {
     }
 
     public get allCards() {
-        return this.state.decklist.allCards.map(this.game.getFromRef);
+        return this.state.decklist.allCards.map((x) => this.game.getFromRef(x));
```
[the remark this answers is not in the document]
Comment (CDMayberry) [reply server/game/core/Player.ts:134]: Hm, this might be a little awkward atm, this is a nested object within state. Need to think about how to handle that. For now leave it as-is.

File: server/utils/SWUStats/SwuStatsHandler.ts [hunk 3]
```diff
@@ -185,6 +185,52 @@ export class SwuStatsHandler {
...
+                played: 0,
+                resourced: 0,
+                activated: 0,
+                drawn: 0,
+                discarded: 0,
+            });
+        }
+
+        for (const cardMetric of statsTracker.cardMetrics) {
+            if (cardMetric.player === player.trackingId && cardResultsByTrackingId.has(cardMetric.card)) {
```
[the remark this answers is not in the document]
Comment (Frugghi) [reply server/utils/SWUStats/SwuStatsHandler.ts:206]: Can do that as well if you prefer! I chose to keep the state simpler because it'll be used more frequently and the array will be iterated over only at the end of the game so it should be fine to be a bit more wasteful.

File: server/game/core/Player.ts [hunk 4]
```diff
@@ -149,6 +150,14 @@ export class Player extends GameObject<IPlayerState> {
         return this._promptState;
     }
 
+    public get trackingId(): string {
+        if (process.env.ENVIRONMENT === 'development') {
```
[the remark this answers is not in the document]
Comment (Frugghi) [reply server/game/core/Player.ts:154]: @moyerr Was this meant to be permanent or was this to simplify the debugging?

I have a feeling it was just to simplify the debugging locally.

File: server/utils/SWUStats/SwuStatsHandler.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (AMMayberry1) [reply server/utils/SWUStats/SwuStatsHandler.ts:206]: Makes sense to me!
